core/src/main/java/io/paradoxical/cassieq/dataAccess/MessageRepositoryImpl.java

```diff
@@ -261,4 +265,26 @@ private void updateQueueSize(int amount){
 
         session.execute(with);
     }
+
+    /**
+     * Determines if the message is allowed to be consumed
+     * @param message
+     * @return
+     */
+    private boolean messageConsumable(final Message message) {
+        if(message.getDeliveryCount() >= queueDefinition.getMaxDeliveryCount()){
```

Comment:
not sure i like this logic in the repo for a consume message method... not that it doesn't look perfect. just not sure how i feel about it.